firewood/src/db.rs

```diff
@@ -436,8 +441,19 @@ pub struct Db {
 impl Db {
     const PARAM_SIZE: u64 = size_of::<DbParams>() as u64;
 
+    #[allow(clippy::new_ret_no_self)]
+    pub async fn new<P: AsRef<Path>>(
+        db_path: P,
+        cfg: &DbConfig,
+    ) -> Result<impl api::Db, api::Error> {
```

Comment:
There shouldn't be a forced indirection here. You should be able to use the concrete database. This is a little awkward.